src/smc-webapp/course/common.cjsx

```diff
@@ -303,7 +304,10 @@ exports.StudentAssignmentInfo = rclass
             error = "Try to #{name.toLowerCase()} again:\n" + error
         <ErrorDisplay key='error' error={error} style={maxHeight: '140px', overflow:'auto'}/>
 
-    render_last: (name, obj, type, enable_copy, copy_tip, open_tip) ->
+    render_last: (name, obj, type, enable_copy, copy_tip, open_tip, opts) ->
```

Comment:
I think this is the first time in all our codebase that opts is used this way.  I would rather you use a different name, e.g., extra_opts, *or*, put all the inputs into opts.  I think everywhere else in the codebase, if opts is used, then it is the only thing used, i.e.,
```
render_last: (opts) ->
    opts = ...
```

Definitely my preference is **very strongly** that if there are so many arguments that there is any possible chance that things could get confusing... then just use opts for all of them.  Combining using opts with a bunch of positional params is asking for trouble.  Positional params are already asking for trouble when we don't have strong typing (like in typescript) or  named params (like in python).

Reply:
It's been used in a few places but usually with less than 3 positional arguments (ie. very few args and the function has very clear default behavior). I think I will convert this over to all opts. It's already hard to map in my head what this random boolean in the middle of the function call is doing.